src/DelegatedRole.php

```diff
@@ -104,6 +104,16 @@ protected static function validate(array $data, Collection $constraints): void
      */
     public function matchesPath(string $target): bool
     {
+        if (isset($this->pathHashPrefixes)) {
+            $targetHash = hash('sha256', $target);
+
+            foreach ($this->pathHashPrefixes as $prefix) {
+                if (str_starts_with($targetHash, $prefix)) {
+                    return true;
```

Comment:
So if we didn't find the tart it would go on to `if ($this->paths) {` but that is ok because our metadata validation ensures they can never be both, correct?

Reply:
Yes.